### PR TITLE
Refs #25217 - Add prop-types to forms

### PR DESCRIPTION
**Pull request by sharvit**

### CODE REVIEW

**Comment by ohadlevy** [comment]
[test foreman]

File: webpack/assets/javascripts/react_app/components/PasswordStrength/__tests__/__snapshots__/PasswordStrength.test.js.snap [hunk 1]
```diff
@@ -3,8 +3,11 @@
 exports[`PasswordStrength component rendering renders password-strength 1`] = `
 <div>
   <CommonForm
+    className=""
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/PasswordStrength/__tests__/__snapshots__/PasswordStrength.test.js.snap:6]: should it be empty?

File: webpack/assets/javascripts/react_app/components/common/forms/Actions.js [hunk 2]
```diff
@@ -1,18 +1,18 @@
 import React from 'react';
+import PropTypes from 'prop-types';
 import { Button } from 'patternfly-react';
+
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/common/forms/Actions.js:4]: nitpick: redundant empty line?

File: webpack/assets/javascripts/react_app/components/common/forms/Actions.js [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
+import { noop } from '../../../common/helpers';
 import { simpleLoader } from '../Loader';
 import { translate as __ } from '../../../../react_app/common/I18n';
 
-export default ({ onCancel, disabled = false, submitting = false }) => (
+const FormActions = ({ onCancel, disabled = false, submitting = false }) => (
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/common/forms/Actions.js]: are these assignments needed here? we have default values via proptypes.

File: webpack/assets/javascripts/react_app/components/common/forms/Actions.js [hunk 4]
```diff
@@ -21,3 +21,17 @@ export default ({ onCancel, disabled = false, submitting = false }) => (
     </div>
   </div>
 );
+
+FormActions.propTypes = {
+  disabled: PropTypes.bool,
+  submitting: PropTypes.bool,
+  onCancel: PropTypes.func,
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/common/forms/Actions.js:28]: shouldn't it be required?

File: webpack/assets/javascripts/react_app/components/common/forms/Checkbox.js [hunk 5]
```diff
@@ -1,12 +1,36 @@
...
+    />
   </CommonForm>
 );
 
+Checkbox.propTypes = {
+  className: PropTypes.string,
+  checked: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
+  label: PropTypes.string,
+  disabled: PropTypes.bool,
+  onChange: PropTypes.func,
```
Comment (amirfefer) [inline webpack/assets/javascripts/react_app/components/common/forms/Checkbox.js:25]: shouldn't it be required as well?

**Review by amirfefer — CHANGES_REQUESTED** [review]
Thanks @sharvit  :+1: 
I wonder if some of the props that you added should be required

File: webpack/assets/javascripts/react_app/components/PasswordStrength/__tests__/__snapshots__/PasswordStrength.test.js.snap (hunk 1 above)
Comment (sharvit) [reply webpack/assets/javascripts/react_app/components/PasswordStrength/__tests__/__snapshots__/PasswordStrength.test.js.snap:6]: Yes!
The prop `className` of `CommonForm` set to be an empty string by default with default-props.

See: https://github.com/theforeman/foreman/pull/6232/files#diff-1c6b849340fffa2e020670491bda3fe4R30

File: webpack/assets/javascripts/react_app/components/common/forms/Actions.js (hunk 2 above)
Comment (sharvit) [reply webpack/assets/javascripts/react_app/components/common/forms/Actions.js:4]: I always split the `3rd-party` modules from the `local-modules` by empty line :smiley:

File: webpack/assets/javascripts/react_app/components/common/forms/Actions.js (hunk 3 above)
Comment (sharvit) [reply webpack/assets/javascripts/react_app/components/common/forms/Actions.js]: Nice catch, forgot to remove them :+1:

File: webpack/assets/javascripts/react_app/components/common/forms/Actions.js (hunk 4 above)
Comment (sharvit) [reply webpack/assets/javascripts/react_app/components/common/forms/Actions.js:28]: The `onCancel`? why?
I can render it with `noop`. IMHO funcs should never be required.

File: webpack/assets/javascripts/react_app/components/common/forms/Actions.js (hunk 3 above)
Comment (sharvit) [reply webpack/assets/javascripts/react_app/components/common/forms/Actions.js]: Fixed

**Comment by sharvit** [comment]
Thanks for the review @amirfefer, can you rereview please?

**Comment by ohadlevy** [comment]
@amirfefer bump?

**Review by amirfefer — APPROVED** [review]
Thanks @sharvit  !
works well :+1:

**Review by ohadlevy — APPROVED** [review]

**Pull request merged**